### PR TITLE
highlight typesem.d error messages

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/typesem.d [hunk 1]
```diff
@@ -205,7 +207,7 @@ private extern (C++) final class TypeSemanticVisitor : Visitor
         mtype.basetype = mtype.basetype.toBasetype().mutableOf();
         if (mtype.basetype.ty != Tsarray)
         {
-            mtype.error(loc, "T in __vector(T) must be a static array, not %s", mtype.basetype.toChars());
+            mtype.error(loc, "T in __vector(T) must be a static array, not `%s`", mtype.basetype.toChars());
```
Comment (JinShil) [inline src/dmd/typesem.d]: Shouldn't `__vector(T)` be surrounded by tick marks also?

File: src/dmd/typesem.d [hunk 2]
```diff
@@ -906,7 +908,7 @@ private extern (C++) final class TypeSemanticVisitor : Visitor
                 {
                     if (ubyte m = fparam.type.mod & (MODimmutable | MODconst | MODwild))
                     {
-                        mtype.error(loc, "cannot have %s out parameter of type %s", MODtoChars(m), t.toChars());
+                        mtype.error(loc, "cannot have %s out parameter of type `%s`", MODtoChars(m), t.toChars());
```
Comment (JinShil) [inline src/dmd/typesem.d]: I think "%s out" should also be surrounded by tick marks:  "cannot have \`%s out\` parameter of type \`%s\`"

File: src/dmd/typesem.d [hunk 3]
```diff
@@ -1062,7 +1064,7 @@ private extern (C++) final class TypeSemanticVisitor : Visitor
 
         if (wildreturn && !wildparams)
         {
-            mtype.error(loc, "inout on return means inout must be on a parameter as well for %s", mtype.toChars());
+            mtype.error(loc, "inout on return means inout must be on a parameter as well for `%s`", mtype.toChars());
```
Comment (JinShil) [inline src/dmd/typesem.d]: Tick marks around "inout" and "return"

File: src/dmd/typesem.d [hunk 4]
```diff
@@ -795,15 +797,15 @@ private extern (C++) final class TypeSemanticVisitor : Visitor
             errors |= tf.checkRetType(loc);
             if (tf.next.isscope() && !(sc.flags & SCOPEctor))
             {
-                mtype.error(loc, "functions cannot return scope %s", tf.next.toChars());
+                mtype.error(loc, "functions cannot return scope `%s`", tf.next.toChars());
```
Comment (JinShil) [inline src/dmd/typesem.d]: Should the tick marks surround "return scope" also? "functions cannot \`return scope %s\`".  I'm not sure.

File: src/dmd/typesem.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/typesem.d]: There are many such decisions, and they don't always have clear answers. In this case, `T` is not a user defined type, it never appears in his source code.

I opted instead to make these changes a bulk "first approximation", taking the most straightforward approach in order to get the bulk of it done.

**Review by JinShil — DISMISSED** [review]

File: src/dmd/typesem.d (hunk 1 above)
Comment (JinShil) [reply src/dmd/typesem.d]: Ok, I understand.

File: src/dmd/typesem.d (hunk 2 above)
Comment (WalterBright) [reply src/dmd/typesem.d]: I opted out of `MODtoChars` results being highlighted for the first approximation. There are thousands more error messages to go.

File: src/dmd/typesem.d (hunk 4 above)
Comment (WalterBright) [reply src/dmd/typesem.d]: Again, this is just a first approximation. There are thousands of messages to fix. The last time I tried to do this, the PRs all got derailed into debates about such things, and the result was zero progress for the following 6 months.

Comment (wilzbach) [reply src/dmd/typesem.d]: > There are thousands of messages to fix. 

Are you sure you don't want to automate this?

**Review by JinShil — APPROVED** [review]

File: src/dmd/typesem.d (hunk 4 above)
Comment (WalterBright) [reply src/dmd/typesem.d]: Others have said they were going to automate it. I'm not waiting any longer.

**Pull request merged**